File: app/src/main/java/com/example/fasst_movie/Adapter/FilmListAdapter.java
package com.example.fasst_movie.Adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.RequestOptions;
import com.example.fasst_movie.Activity.DetailActivity;
import com.example.fasst_movie.Domain.Film;
import com.example.fasst_movie.R;

import java.util.ArrayList;

public class FilmListAdapter extends RecyclerView.Adapter<FilmListAdapter.Viewholder> {
    ArrayList<Film> items;
    Context context;

    public FilmListAdapter(ArrayList<Film> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public FilmListAdapter.Viewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        View inflate = LayoutInflater.from(parent.getContext()).inflate(R.layout.film_viewholder, parent, false);
        return new Viewholder(inflate);
    }

    @Override
    public void onBindViewHolder(@NonNull FilmListAdapter.Viewholder holder, int position) {
        holder.titleTxt.setText(items.get(position).getTitle());
        RequestOptions requestOptions = new RequestOptions();
        requestOptions = requestOptions.transform(new CenterCrop(), new RoundedCorners(30));

        Glide.with(context)
                .load(items.get(position).getPoster())
                .apply(requestOptions)
                .into(holder.pic);

        holder.itemView.setOnClickListener(v -> {
                Intent intent = new Intent(context, DetailActivity.class);
                intent.putExtra("object",items.get(position));
                context.startActivity(intent);

        });

    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class Viewholder extends RecyclerView.ViewHolder {
        TextView titleTxt;
        ImageView pic;
        public Viewholder(@NonNull View itemView) {
            super(itemView);
            titleTxt = itemView.findViewById(R.id.nameText);
            pic = itemView.findViewById(R.id.pic);
        }
    }
}

File: app/src/main/java/com/example/fasstappbookingtic/Adapter/FilmListAdapter.kt
package com.example.fasstappbookingtic.Adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.example.fasstappbookingtic.Activity.FilmDetailActivity
import com.example.fasstappbookingtic.Models.Film
import com.example.fasstappbookingtic.Models.SliderItems
import com.example.fasstappbookingtic.databinding.ViewholderFilmBinding

class FilmListAdapter(private val items: ArrayList<Film>) :RecyclerView.Adapter<FilmListAdapter.Viewholder>() {
    private var context:Context? = null

    inner class Viewholder(private val binding: ViewholderFilmBinding):RecyclerView.ViewHolder(binding.root) {

        fun bind(film:Film) {
            binding.nameTxt.text = film.Title
            val requestOptions = RequestOptions()
                .transform(CenterCrop(),RoundedCorners(30))

            Glide.with(context!!)
                .load(film.Poster)
                .apply(requestOptions)
                .into(binding.pic)

            binding.root.setOnClickListener {
                val intent = Intent(context,FilmDetailActivity::class.java)
                intent.putExtra("object",film)
                context!!.startActivity(intent)
            }
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilmListAdapter.Viewholder {
        context = parent.context
        val binding = ViewholderFilmBinding.inflate(LayoutInflater.from(context),parent,false)
        return Viewholder(binding)
    }

    override fun onBindViewHolder(holder: FilmListAdapter.Viewholder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int = items.size

}











